### ISSUE TITLE
[Bug]: HasManyThrough relation

### ISSUE THREAD

**Issue by brendt** [tempestphp/tempest-framework#497]
### Tempest Version

1.0-alpha1

### PHP Version

8.3

### Operating System

MacOS

### Description

Tempest doesn't have a shorthand for HasManyThrough relation (eg. User - UserPermission - Permission), which is fine. However, manually creating this relation doesn't work either. 

### Steps to Reproduce

```php
final readonly class CreatePermissionsTable implements Migration
{
    public function getName(): string
    {
        return '0000-00-01_create_permissions_table';
    }

    public function up(): CreateTableStatement
    {
        return (new CreateTableStatement('permissions'))
            ->primary()
            ->varchar('name');
    }

    public function down(): DropTableStatement
    {
        return DropTableStatement::forModel(Permission::class);
    }
}

final readonly class CreateUserPermissionTable implements Migration
{
    public function getName(): string
    {
        return '0000-00-02_create_user_permissions_table';
    }

    public function up(): CreateTableStatement
    {
        return (new CreateTableStatement('user_permissions'))
            ->primary()
            ->belongsTo('user_permissions.user_id', 'users.id')
            ->belongsTo('user_permissions.permission_id', 'permissions.id');
    }

    public function down(): DropTableStatement
    {
        return DropTableStatement::forModel(Permission::class);
    }
}

final readonly class CreateUsersTable implements Migration
{
    public function getName(): string
    {
        return '0000-00-00_create_users_table';
    }

    public function up(): CreateTableStatement
    {
        return (new CreateTableStatement('users'))
            ->primary()
            ->varchar('name')
            ->varchar('email')
            ->datetime('emailValidatedAt', nullable: true)
            ->text('password');
    }

    public function down(): DropTableStatement
    {
        return DropTableStatement::forModel(User::class);
    }
}

final class Permission implements DatabaseModel
{
    use IsDatabaseModel;

    public function __construct(
        public string $name,
    ) {}
}

final class UserPermission implements DatabaseModel
{
    use IsDatabaseModel;

    public function __construct(
        public User $user,
        public Permission $permission,
    ) {}
}


final class User implements DatabaseModel, CanAuthenticate, CanAuthorize
{
    use IsDatabaseModel;

    public string $password;

    public function __construct(
        public string $name,
        public string $email,
        /** @var \Tempest\Auth\UserPermission[] $userPermissions */
        public array $userPermissions = [],
    ) {
    }

    /**
     * @param string $password The raw password, which will be encrypted as soon as it is set
     */
    public function setPassword(#[SensitiveParameter] string $password): self
    {
        $this->password = password_hash($password, PASSWORD_BCRYPT);

        return $this;
    }

    public function getPermissions(): array
    {
        return $this->permissions;
    }

    public function hasPermission(UnitEnum|string $permission): bool
    {
        return arr($this->permissions)->contains($permission);
    }
}
```

```php

        $user = (new User(
            name: 'Brent',
            email: 'brendt@stitcher.io',
        ))
            ->setPassword('password')
            ->save();

        $permission = (new Permission(name: 'admin'))->save();

        $userPermission = (new UserPermission(
            user: $user,
            permission: $permission
        ))->save();

        $user->load('userPermissions.permission');
```

**Comment by brendt** [comment]
Fixed with #493

**Comment by blackshadev** [comment]
@brendt #493 is merged so is this already fixed?
